### PR TITLE
reformat layer norms

### PR DESCRIPTION
**Pull request by cybermind-nick**
- change to layer_norm
- reformated layer_norms in '/ivy/functional/' to standard
- fixed creation formatting
- fixed functional/norm.py formatting
- fixed test_core/test_creation.py formatting
- fixed functional/norm.py docstring formatting
- fixed issue with typing.Union

### CODE REVIEW

**Comment by cybermind-nick** [comment]
#778

**Comment by cybermind-nick** [comment]
#1168

**Comment by djl11** [comment]
Hi @cybermind-nick, very sorry it's been so long to respond to your PR. We will get this reviewed right away!

**Comment by bicycleman15** [comment]
Hey, it's a good start, however, there are still some change are required.
I would like you to go through the following guideline:
https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html

The examples are good, but they just need to be formatted according to the above guideline. 
Let me know when you are done with these changes. Also feel free to ask me any questions 🙂

**Review by bicycleman15 — CHANGES_REQUESTED** [review]
Please read my comment

**Comment by cybermind-nick** [comment]
Thanks for the review  @bicycleman15. 

I've reread the guide and I think my only omission was the '\' at the end of my multiline variables in the docstring examples.
Is that right or did I miss something else?

**Comment by cybermind-nick** [comment]
I would as like to point out that the `ivy.Container` example was unsuccessful because it kept throwing an 'index out of bounds' exception everytime I passed it into the ivy.layer_norms() function. So I included the syntax structure of the ivy.Container layer_norm() example, but not the output.

This is the error shown below:
`Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/ivy/ivy/func_wrapper.py", line 78, in _method_wrapped
    return f(*args, out=out, **kwargs)
  File "/ivy/ivy/container/wrapping.py", line 9, in new_fn
    data_idx = ivy.__dict__[fn_name].array_spec[0]
IndexError: list index out of range`

If there is a fix I'm missing or a need to remove that example, please let me know.

Every other guideline rule I've observed has been followed. If there is any I'm missing, please let me know too.

Thanks

**Comment by cybermind-nick** [comment]
Hello 👋🏽 @bicycleman15

I already made the changes and requested for review. It's been a while now.
Feedback on my work will be much appreciated.

**Comment by cybermind-nick** [comment]
Hello 👋🏽 @djl11

It would seem my reviewer is unavailable. Can I just submit the present link?

File: ivy/functional/ivy/norms.py [hunk 1]
```diff
@@ -28,8 +37,46 @@ def layer_norm(x, normalized_idxs, epsilon=None, scale=None, offset=None, new_st
 
     Returns
     -------
-    ret
+     ret
         The layer after applying layer normalization.
+    
+    Example
+    -------
+
```
Comment (bicycleman15) [inline ivy/functional/ivy/norms.py:45]: You need to put headers like following as mentioned in the contributor guidelines (https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html):

```
Functional Examples
-------------------

With :code:`ivy.Array` input:
```

File: ivy/functional/ivy/norms.py [hunk 2]
```diff
@@ -28,8 +37,46 @@ def layer_norm(x, normalized_idxs, epsilon=None, scale=None, offset=None, new_st
...
+        [ 0.1047,  0.5886,  1.2732], \
+        [ 0.7696, -1.7024, -2.2518]])
+    >>> norm = ivy.layer_norm(arr, [0, 1, 2], epsilon=0.001, \
+                 new_std=1.5, offset=0.5, scale=0.5)
+    >>> print(norm)
+    ivy.array([[ 0.57629204,  0.29217   ,  1.3311275 ],
+           [ 0.58084774,  0.89134157,  1.3306142 ],
+           [ 1.0074799 , -0.5786756 , -0.9311974 ]], dtype=float32)
+
+    >>> import tensorflow as tf # ivy.NativeArray
```
Comment (bicycleman15) [inline ivy/functional/ivy/norms.py:68]: You don't need to import tensorflow. You can create a new instance of native array using `ivy.native_array`
For eg.
```
With :code:`ivy.NativeArray` input:

>>> x = ivy.native_array([0, 1, 2])
>>> y = ivy.tan(x)
>>> print(y)
ivy.array([0., 1.5574077, -2.1850398])
```
Above is mentioned in the guidelines (https://lets-unify.ai/ivy/deep_dive/11_docstring_examples.html)

File: ivy/functional/ivy/norms.py [hunk 3]
```diff
@@ -28,8 +37,46 @@ def layer_norm(x, normalized_idxs, epsilon=None, scale=None, offset=None, new_st
...
+    >>> ivy.set_backend('tensorflow')
+    >>> tensor = tf.constant([[3.,1.],[4.,12.]])
+    >>> norm = ivy.layer_norm(tensor, [0,1], new_std=1.25, offset=0.25, scale=0.3)
+    >>> print(norm)
+    ivy.array([[ 0.07071576, -0.10856849],
+           [ 0.16035788,  0.8774949 ]], dtype=float32)
+
+    >>> container = ivy.Container({'a': ivy.array([2.,3.,4.]),
```
Comment (bicycleman15) [inline ivy/functional/ivy/norms.py:76]: I will have to check and get back to you on this.

**Review by bicycleman15 — CHANGES_REQUESTED** [review]

**Comment by bicycleman15** [comment]
Regarding the error above, @sherry30 could you have a look at it ? I think `ivy.layer_norms` should be nestable.

**Comment by VedPatwardhan** [comment]
> Regarding the error above, @VedPatwardhan could you have a look at it? I think `ivy.layer_norm` should be nestable.

Sure @bicycleman15, I'll take a look and get back at the earliest!

File: ivy/functional/ivy/norms.py (hunk 2 above)
Comment (cybermind-nick) [reply ivy/functional/ivy/norms.py:68]: Hiya 👋🏽 @bicycleman15 . Thanks for the review

Using `ivy.native_array` was my first thought. I tried it with three backends (tensorflow, torch and numpy) and they all threw No Attribute errors. That's why I defaulted to using an actual framework array to show that the function work with the native arrays of the wrapped frameworks and was agnostic

**Below are the AttributeErrors from all 3 backends:**
Tensorflow:
` arr = ivy.native_array([0.1, 2.1, 3.])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'ivy.functional.backends.tensorflow' has no attribute 'native_array'`

Torch:
`arr = ivy.native_array([0.1, 2.1, 3.])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'ivy.functional.backends.torch' has no attribute 'native_array'`

Numpy:
`arr = ivy.native_array([0.1, 2.1, 3.])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'ivy.functional.backends.numpy' has no attribute 'native_array'`


I hope that clears things up on my reasons. Thanks

**Comment by cybermind-nick** [comment]
@bicycleman15 

I've pushed the change to the Header. Thank you for the correction

File: ivy/functional/ivy/norms.py (hunk 2 above)
Comment (bicycleman15) [reply ivy/functional/ivy/norms.py:72]: Ah okay.

Maybe for now, let's ignore this error. It is something that we will have to fix. 🙂

Comment (bicycleman15) [reply ivy/functional/ivy/norms.py:72]: I would still say make it `ivy.native_array` only. Don't worry about the failing test.

**Review by bicycleman15 — CHANGES_REQUESTED** [review]

**Comment by bicycleman15** [comment]
Rest LGTM! I will merge once you make these above changes.

**Comment by cybermind-nick** [comment]
Hey @bicycleman15 , I've made the switch to `ivy.native_array`

Thanks for all the help

**Review by bicycleman15 — APPROVED** [review]

**Pull request merged**